assets/src/edit-story/components/library/panes/media/mediaElement.js

```diff
@@ -125,8 +126,14 @@ const MediaElement = ({
     sizes,
     local,
   } = resource;
-  const oRatio =
+
+  let oRatio =
     originalWidth && originalHeight ? originalWidth / originalHeight : 1;
+
+  if (hasGeneratedPoster(resource)) {
```

Comment:
Could you pls clarify this? You're using the dimensions of the generated poster image to override the video dimensions that are in the `resource` object? If so, I don't think this code belongs here. We want to do this override in the `Resource` object as early as possible. In the media reducers or the `getResourceFrom*` method. Otherwise this could would be duplicating throughout.

Reply:
I can't live in `getResourceFrom` as these values are updated in client when a poster is generated.